code/iaas/resource-pool/src/main/java/io/cattle/platform/resource/pool/impl/ResourcePoolManagerImpl.java

```diff
@@ -148,14 +153,23 @@ protected String getItem(Map<Object, Object> keys, Object pool, String qualifier
             return null;
         }
 
+        // trying to allocate resource from tryItems
+        for (int i = 0; i < tryItems.size(); i++) {
+            if (tryItems.get(i) != null && generator.isInPool(tryItems.get(i))) {
```

Comment:
We can also validate against the string being not null, but empty here. You can cover both cases by using !StringUtils.isEmpty() clause